### PR TITLE
Api/manage tokens

### PR DESCRIPTION
**Pull request by imnotashrimp**
# What changed

- added log shipping token endpoints
- moved /v1 from baseurl to each endpoint in prep for /v2 endpoints

----

## Pages to review

- https://deploy-preview-398--logz-docs.netlify.app/api/#tag/Manage-log-shipping-tokens

## Remaining work

- [ ] Technical review - this needs to be dev, and probably also support
- [ ] Copy Review
- [ ] @shalper Test all code samples. You'll need to request that the feature flag gets enabled on your account.

## Post launch

To be completed by the docs team upon merge:

- [ ] Announce the change in the Support slack channel
- [ ] @shalper Open a new PR for updated "tokens" docs - this is going to affect shared tokens and API tokens, because the UI is changing. This is a lower priority than the API doc, so leave this until later today or early next week.

### CODE REVIEW

File: _source/api/logzio-public-api.yml [hunk 1]
```diff
@@ -103,6 +102,10 @@ tags:
   - name: Manage shared tokens
     description: >-
       **Note:** Shared token endpoints require permissions that must be set by our Support team. Please email help@logz.io for assistance.
+  - name: Manage log shipping tokens
+    description: >-
+      Use these API requests to create, update, retrieve, or delete shipping tokens.
+      Data shipped using this token will be indexed under the account associated with this ID.
```
Comment (imnotashrimp) [inline _source/api/logzio-public-api.yml]: @shalper "Data shipped using this token" - In this case, we're not referring to a specific token.

File: _source/api/logzio-public-api.yml [hunk 2]
```diff
@@ -760,8 +764,178 @@ paths:
...
+    post:
+      tags:
+        - Manage log shipping tokens
+      summary: Create a log shipping token
+      description: Creates a log shipping token for this account.
+      operationId: createNewLogShippingToken
+      x-code-samples:
+        - lang: cURL
+          source: >-
+            curl --location --request GET 'https://api.logz.io/v1/account-management/whoami/' \
```
Comment (imnotashrimp) [inline _source/api/logzio-public-api.yml]: This feature needs to be enabled somewhere so you can run test calls with the code samples. Please look into this in the morning.

File: _source/api/logzio-public-api.yml [hunk 3]
```diff
@@ -760,8 +764,178 @@ paths:
...
+      responses:
+        '200':
+          description: successful operation
+          schema:
+            $ref: '#/definitions/ShippingTokensModel'
+    delete:
+      tags:
+        - Manage log shipping tokens
+      summary: Delete a log shipping token
+      description: # TODO
```
Comment (imnotashrimp) [inline _source/api/logzio-public-api.yml]: - [ ] Description needs to be written @shalper

File: _source/api/logzio-public-api.yml [hunk 4]
```diff
@@ -760,8 +764,178 @@ paths:
...
+        '200':
+          description: successful operation
+          schema:
+            $ref: '#/definitions/ShippingTokensLimitsResponse'
+  /v1/log-shipping/tokens/search:
+    post:
+      tags:
+        - Manage log shipping tokens
+      summary: Retrieve log shipping tokens
+      description: # TODO
```
Comment (imnotashrimp) [inline _source/api/logzio-public-api.yml]: - [ ] description needs to be written @shalper

**Comment by imnotashrimp** [comment]
@boofinka Those are excellent points. @shalper this should go in the description at the top of the feature, and probably some shorter version in each of the requests for DELETING and UPDATING.

Re: the last shipping token, the core of this feature is that you have to have at least one. @shalper 
 can double-check with dev and test once the api goes live.

File: _source/api/logzio-public-api.yml (hunk 2 above)
Comment (shalper) [reply _source/api/logzio-public-api.yml]: This feature is still not deployed. Waiting for dev.

File: _source/api/logzio-public-api.yml (hunk 3 above)
Comment (shalper) [reply _source/api/logzio-public-api.yml]: Done

File: _source/api/logzio-public-api.yml (hunk 4 above)
Comment (shalper) [reply _source/api/logzio-public-api.yml]: Done

**Comment by shalper** [comment]
> I think we should have a big honkin' warning saying that changing or deleting a token means it cannot be used in shipping, effective immediately; maybe go as far as suggest that the user double-checks that all running shippers have been updated with a different token.
> 
> Also, what happens if someone tries to delete the only shipping token an account has? If we allow it, this is another place where a warning is due.

Checked.  The last token can't be deleted. Added your tip directly in the docs. Thanks!

File: _source/api/logzio-public-api.yml [hunk 5]
```diff
@@ -760,8 +762,181 @@ paths:
...
+      responses:
+        '200':
+          description: successful operation
+          schema:
+            $ref: '#/definitions/ShippingTokensModel'
+    put:
+      tags:
+        - Manage log shipping tokens
+      summary: Update a log shipping token
+      description: Changes the name of a log shipping token or switches status between enabled/disabled.
```
Comment (renanaWasserm) [inline _source/api/logzio-public-api.yml]: the user can edit the name and disable/enable the token in the same request

File: _source/api/logzio-public-api.yml [hunk 6]
```diff
@@ -760,8 +762,181 @@ paths:
...
+          description: successful operation
+          schema:
+            $ref: '#/definitions/ShippingTokensModel'
+  /v1/log-shipping/tokens/limits:
+    get:
+      tags:
+        - Manage log shipping tokens
+      summary: Get number of available tokens
+      description: >-
+        Returns the number of log shipping tokens that can be enabled.
```
Comment (renanaWasserm) [inline _source/api/logzio-public-api.yml]: it returns the number of log shipping tokens that can be enabled AND the number of currently enabled tokens

File: _source/api/logzio-public-api.yml [hunk 7]
```diff
@@ -760,8 +762,184 @@ paths:
...
+    post:
+      tags:
+        - Manage log shipping tokens
+      summary: Create a log shipping token
+      description: Creates a log shipping token for this account.
+      operationId: createNewLogShippingToken
+      x-code-samples:
+        - lang: cURL
+          source: >-
+            curl --location --request GET 'https://api.logz.io/v1/account-management/whoami/' \
```
Comment (imnotashrimp) [inline _source/api/logzio-public-api.yml]: I made the wrong endpoint in this code sample. Needs to be fixed @shalper

File: _source/api/logzio-public-api.yml [hunk 8]
```diff
@@ -760,8 +762,184 @@ paths:
...
+      responses:
+        '200':
+          description: successful operation
+          schema:
+            $ref: '#/definitions/ShippingTokensModel'
+    put:
+      tags:
+        - Manage log shipping tokens
+      summary: Update a log shipping token
+      description: Changes the name of a log shipping token or switches status between enabled/disabled.
```
Comment (shalper) [inline _source/api/logzio-public-api.yml:820]:
```suggestion
      description: Enables/disables a log shipping token and/or renames it.
```

File: _source/api/logzio-public-api.yml [hunk 9]
```diff
@@ -760,8 +762,184 @@ paths:
...
+          description: successful operation
+          schema:
+            $ref: '#/definitions/ShippingTokensModel'
+  /v1/log-shipping/tokens/limits:
+    get:
+      tags:
+        - Manage log shipping tokens
+      summary: Get number of available tokens
+      description: >-
+        Returns the number of log shipping tokens that can be enabled.
```
Comment (shalper) [inline _source/api/logzio-public-api.yml:887]:
```suggestion
        Returns the number of log shipping tokens currently in use and the number of available tokens that can be enabled.
```

File: _source/api/logzio-public-api.yml (hunk 7 above)
Comment (renanaWasserm) [inline _source/api/logzio-public-api.yml:776]: The URL here is wrong

File: _source/api/logzio-public-api.yml [hunk 10]
```diff
@@ -760,8 +762,184 @@ paths:
...
+          example: 786351
+        - in: body
+          name: body
+          required: false
+          schema:
+            $ref: '#/definitions/ShippingTokensRequest'
+      x-code-samples:
+        - lang: cURL
+          source: >-
+            curl -L -X PUT 'https://api.logz.io/v1/log-shipping/tokens/1900' \
```
Comment (renanaWasserm) [inline _source/api/logzio-public-api.yml:838]: 1900 -> {id} as written in the description of get endpoint

File: _source/api/logzio-public-api.yml [hunk 11]
```diff
@@ -760,8 +762,184 @@ paths:
...
+        - name: id
+          in: path
+          required: true
+          type: integer
+          format: int32
+          description: This token's ID.
+          example: 786351
+        - in: body
+          name: body
+          required: false
```
Comment (renanaWasserm) [inline _source/api/logzio-public-api.yml:832]: the body is required

File: _source/api/logzio-public-api.yml [hunk 12]
```diff
@@ -760,8 +762,184 @@ paths:
...
+            curl -L -X POST 'https://api.logz.io/v1/log-shipping/tokens/search' \
+            -H 'X-API-TOKEN: <token>' \
+            -H 'Content-Type: application/json' \
+            --data-raw '{
+              "filter": {
+                "enabled": true
+              },
+              "sort": [
+                {
+                  "field": "NAME",
```
Comment (renanaWasserm) [inline _source/api/logzio-public-api.yml:926]: NAME -> name (the sort-by values should be as the parameter names)

File: _source/api/logzio-public-api.yml [hunk 13]
```diff
@@ -2832,6 +3010,134 @@ definitions:
...
+        example: shalom.the.mighty@gmail.com
+      createdAt:
+        type: integer
+        format: int64
+        description: Unix timestamp of when this token was created.
+        example: 389836800
+      createdBy:
+        type: string
+        description: Email address of the user who created this token.
+        example: you.got.this.sara@gmail.com
```
Comment (renanaWasserm) [inline _source/api/logzio-public-api.yml:3047]: do you want to keep it? :)

File: _source/api/logzio-public-api.yml [hunk 14]
```diff
@@ -2832,6 +3010,134 @@ definitions:
...
+  ShippingTokensSortRequest:
+      type: object
+      properties:
+        field:
+          type: string
+          enum:
+            - CREATED_AT
+            - NAME
+          description: >-
+            To sort by creation date, use `"CREATED_AT"`.
```
Comment (renanaWasserm) [inline _source/api/logzio-public-api.yml:3095]: createdAt, name

File: _source/api/logzio-public-api.yml (hunk 13 above)
Comment (shalper) [inline _source/api/logzio-public-api.yml:3047]:
```suggestion
        example: you.got.this@gmail.com
```

File: _source/api/logzio-public-api.yml (hunk 14 above)
Comment (shalper) [inline _source/api/logzio-public-api.yml:3095]:
```suggestion
            To sort by creation date, use `"createdAt"`.
```

File: _source/api/logzio-public-api.yml (hunk 13 above)
Comment (shalper) [inline _source/api/logzio-public-api.yml:3047]:
```suggestion
        example: you.got.this@gmail.com
```

File: _source/api/logzio-public-api.yml [hunk 15]
```diff
@@ -2832,6 +3010,134 @@ definitions:
...
+            To sort by name, use `"NAME"`.
```
Comment (shalper) [inline _source/api/logzio-public-api.yml:3096]:
```suggestion
            To sort by name, use `"name"`.
```

File: _source/api/logzio-public-api.yml (hunk 12 above)
Comment (shalper) [inline _source/api/logzio-public-api.yml:926]:
```suggestion
                  "field": "name",
```

File: _source/api/logzio-public-api.yml (hunk 11 above)
Comment (shalper) [reply _source/api/logzio-public-api.yml:832]: fixed

File: _source/api/logzio-public-api.yml (hunk 13 above)
Comment (shalper) [reply _source/api/logzio-public-api.yml]: fixed

File: _source/api/logzio-public-api.yml [hunk 16]
```diff
@@ -760,8 +762,185 @@ paths:
...
+          example: 786351
+        - in: body
+          name: body
+          required: false
+          schema:
+            $ref: '#/definitions/ShippingTokensRequest'
+      x-code-samples:
+        - lang: cURL
+          source: >-
+            curl -L -X PUT 'https://api.logz.io/v1/log-shipping/tokens/1900' \
```
Comment (shalper) [inline _source/api/logzio-public-api.yml:839]:
```suggestion
            curl -L -X PUT 'https://api.logz.io/v1/log-shipping/tokens/{ID}' \
```

**Pull request closed without merging**